### PR TITLE
1500 dev bms query agent

### PR DESCRIPTION
**Pull request by sandradeng20**
This agent is built to query zones information and lab equipment from lab, bms and ArduinoMeterForCaresLabFridge01 namespace. 

Integration test is ignored in the code, because it causes error when building the docker image, but it can pass with mvn test.

### CODE REVIEW

**Review by csl37 — APPROVED** [review]
- remove snapshot versions in pom, dockerfile etc.
- 2 TODOs in the README

**Comment by sandradeng20** [comment]
Remove unnecessary CORS code in BMSQueryLauncher, and override processRequestParameters instead of doGet

File: Agents/BMSQueryAgent/README.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+repo_username.txt should contain your github username, and repo_password.txt your github [personal access token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token),
+which must have a 'scope' that [allows you to publish and install packages](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry#authenticating-to-github-packages).
+
+Then build image with:
+```
+docker build . -t bms-query-agent:1.0.0
+```
+
+### 2) Add Config to Stack Manager
+Copy `stack-manager-input-config-service/bms-query-agent.json` to `TheWorldAvatar/Deploy/dynamic/stack-manager/inputs/config/services/`.
```
Comment (csl37) [inline Agents/BMSQueryAgent/README.md:45]: Deploy/stacks/dynamic

Comment (csl37) [reply Agents/BMSQueryAgent/README.md:45]: Lines 45 and 47

File: Agents/BMSQueryAgent/BMSQueryAgentCode/src/main/java/uk/ac/cam/cares/jps/agent/bmsquery/BMSQueryAgentLauncher.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+ */
+@WebServlet(urlPatterns = {"/status", "/retrieve/equipment", "/retrieve/zones"})
+public class BMSQueryAgentLauncher extends JPSAgent {
+    private static final Logger LOGGER = LogManager.getLogger(BMSQueryAgentLauncher.class);
+
+    static final String KEY_ROOMIRI = "roomIRI";
+
+    public static final String PARAMETERS_VALIDATION_ERROR_MSG = "Unable to validate request sent to the agent.";
+    public static final String EMPTY_PARAMETER_ERROR_MSG = "Empty Request.";
+    public static final String AGENT_Construction_ERROR_MSG = "The BMSQueryAgent could not be constructed.";
```
Comment (sganguly-cares) [inline Agents/BMSQueryAgent/BMSQueryAgentCode/src/main/java/uk/ac/cam/cares/jps/agent/bmsquery/BMSQueryAgentLauncher.java:31]: Capitalise "Construction" for consistency

Comment (sganguly-cares) [inline Agents/BMSQueryAgent/BMSQueryAgentCode/src/main/java/uk/ac/cam/cares/jps/agent/bmsquery/BMSQueryAgentLauncher.java:31]: Capitalise "Construction" for consistency

**Pull request merged**